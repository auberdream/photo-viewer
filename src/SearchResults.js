import React, { Component } from 'react';
import Images from './resources/images';
import Styles from './resources/styles/search-results.css';

class SearchResults extends Component {
  render() {
    return (
      <div className="searchresults-container">
        { this.props.currentSearch.map((item, index) =>
          <div className="search-result" key={ index } onClick={ () => this.props.handler(item.id) }>
            <img className="search-img" src={ item.url } />
            <span>{ item.first_name + " " + item.last_name }</span>
          </div>
        )}
      </div>
    )
  }
}

export default SearchResults;
