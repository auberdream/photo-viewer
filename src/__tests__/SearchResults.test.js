import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from '../SearchResults';

describe('SearchResults', () => {

  const dummyCurrentSearch = [
                                {
                                  "id": 1,
                                  "first_name": "Alan",
                                  "last_name": "Berryhill",
                                  "url": "http://fakeyfake.com"
                                },
                                {
                                  "id": 2,
                                  "first_name": "Allyson",
                                  "last_name": "Brar",
                                  "url": "http://www.noodles.com"
                                },
                                {
                                  "id": 3,
                                  "first_name": "Alvera",
                                  "last_name": "Tinker",
                                  "url": "http://fakeurl.com"
                                }
                              ]
  const wrapper = shallow(<SearchResults currentSearch={ dummyCurrentSearch } handler={Function} />)

  it('renders the images', () => {
    const images = wrapper.find('img');
    expect(images.length).toEqual(3)
  })

  it('renders the names of the photographers', () => {
    const firstName = wrapper.find('span').at(0)
    const secondName = wrapper.find('span').at(1)
    const thirdName = wrapper.find('span').at(2)
    expect(firstName.text()).toEqual('Alan Berryhill')
    expect(secondName.text()).toEqual('Allyson Brar')
    expect(thirdName.text()).toEqual('Alvera Tinker')
  })

})
