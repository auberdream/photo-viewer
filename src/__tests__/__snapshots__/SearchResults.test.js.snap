// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<div
  className="searchresults-container"
>
  <div
    className="search-result"
    onClick={[Function]}
  >
    <img
      className="search-img"
      src="http://fakeyfake.com"
    />
    <span>
      Alan Berryhill
    </span>
  </div>
  <div
    className="search-result"
    onClick={[Function]}
  >
    <img
      className="search-img"
      src="http://www.noodles.com"
    />
    <span>
      Allyson Brar
    </span>
  </div>
  <div
    className="search-result"
    onClick={[Function]}
  >
    <img
      className="search-img"
      src="http://fakeurl.com"
    />
    <span>
      Alvera Tinker
    </span>
  </div>
</div>
`;
